api/services/CasebookService.js

```diff
@@ -1,53 +1,65 @@
 const crypto = require('crypto');
-const apiQueryString = require('querystring');
-const request = require('request');
-const requestPromise = require('request-promise');
-const sails = require('sails');
-
-const CasebookService = {
-    get(options) {
-        return CasebookService._returnRequestMethods('get', options);
-    },
+const axios = require('axios');
+const https = require('https');
+
+const config = require('../../config/environment-variables')
+const {
+    hmacKey,
+    casebookCertificate: cert,
+    casebookKey: key,
+    customURLs,
+} = config;
 
-    post(options) {
-        return CasebookService._returnRequestMethods('post', options);
+const CasebookService = axios.create({
+    baseURL: customURLs.casebookBaseUrl,
+    headers: {
+        'Content-Type': 'application/json; charset=utf-8',
+        'api-version': '4',
     },
+    httpsAgent: new https.Agent({
+        cert,
+        key,
+        keepAlive: true,
+    }),
+    paramsSerializer: queryParamObjToStr,
+    transformRequest: [addHmacToQueryParam],
+});
 
-    _returnRequestMethods(method, options) {
-        const requestType = options.promise ? requestPromise : request;
-        delete options.promise;
+function queryParamObjToStr(queryParamsObj) {
+    const params = new URLSearchParams();
 
-        const authParams = CasebookService._createAuthParams(options.qs);
-        const optionsWithAuthParams = { ...options, ...authParams };
+    Object.entries(queryParamsObj).forEach(([key, value]) => {
```

Comment:
Good point. The service was using queryString before but apparently, it's depreciated so I got rid of it. And it looks like `URLSearchParams` doesn't have a native way of parsing arrays.

<img width="1029" alt="Screenshot 2021-11-11 at 10 05 40" src="https://user-images.githubusercontent.com/1377253/141279182-0590ab3f-5bb4-4f2d-a06f-19d317d39113.png">
e

Reply:
oops, just read the docs, yeah - it is now legacy and not recommended but it's not deprecated. 

have you had a look at URLSearchParams on the node docs? https://nodejs.org/api/url.html#new-urlsearchparamsobj

Reply:
Yeah, I saw that but annoyingly it doesn't quite do what Casebook wants. The code snippet you posted returns this:
```
'user=abc&query=first%2Csecond'
```

But this is what Casebook wants.
```
user=abc&query=first&query=second'
```

Not sure if `URLSearchParams` has an option for this but I wasn't able to find one.

Reply:
the documentation says this:
> Arrays are stringified using array.toString(), which simply joins all array elements with commas.

so what's happening is it's actually trying to encode `{ query: 'first,second' }` 

```.js
// Using an array
const params = new URLSearchParams([
  [ 'user',  'abc' ]
  ['query', 'first'],
  ['query', 'second'],
]);
console.log(params.toString());
// Prints 'user=abc&query=first&query=second'
```
I suppose if you'd prefer to pass in an object rather than tuples, your method is all good 👍

Reply:
Nah, it's more work than it's worth. I need the object to calculate the hash. I think my function is more appropriate in this specific scenario but this is useful to know for the future.